mlflow/entities/webhook.py

```diff
@@ -0,0 +1,170 @@
+class WebhookEvent(str, Enum):
+    # Registered Model Events
+    REGISTERED_MODEL_CREATED = "REGISTERED_MODEL_CREATED"
+
+    # Model Version Events
+    MODEL_VERSION_CREATED = "MODEL_VERSION_CREATED"
+    MODEL_VERSION_TAG_SET = "MODEL_VERSION_TAG_SET"
+    MODEL_VERSION_TAG_DELETED = "MODEL_VERSION_TAG_DELETED"
+
+    # Model Version Alias Events
+    MODEL_VERSION_ALIAS_CREATED = "MODEL_VERSION_ALIAS_CREATED"
+    MODEL_VERSION_ALIAS_DELETED = "MODEL_VERSION_ALIAS_DELETED"
+
+    @classmethod
+    def from_proto(cls, proto: int) -> "WebhookEvent":
+        return cls(ProtoWebhookEvent.Name(proto))
+
+    def to_proto(self) -> int:
+        return ProtoWebhookEvent.Value(self.value)
+
+
+class Webhook:
+    """
+    MLflow entity for Model Registry Webhook.
+
+    A webhook represents a configuration for receiving notifications
+    about model registry events via HTTP POST requests.
+    """
+
+    def __init__(
+        self,
+        webhook_id: str,
+        name: str,
+        url: str,
+        events: list[Union[str, WebhookEvent]],
+        creation_timestamp: int,
+        last_updated_timestamp: int,
+        description: Optional[str] = None,
+        status: Union[str, WebhookEvent] = WebhookStatus.ACTIVE,
+        secret: Optional[str] = None,
+    ):
+        """
+        Initialize a Webhook entity.
+
+        Args:
+            webhook_id: Unique webhook identifier
+            name: Human-readable webhook name
+            url: Webhook endpoint URL
+            events: List of event types that trigger this webhook (strings or WebhookEvent enums)
+            creation_timestamp: Creation timestamp in milliseconds since Unix epoch
+            last_updated_timestamp: Last update timestamp in milliseconds since Unix epoch
+            description: Optional webhook description
+            status: Webhook status (ACTIVE or DISABLED)
+            secret: Optional secret key for HMAC signature verification
+        """
+        super().__init__()
+        self._webhook_id = webhook_id
+        self._name = name
+        self._url = url
+        if not events:
+            raise MlflowException.invalid_parameter_value("Webhook events cannot be empty")
+        self._events = [(WebhookEvent(e) if isinstance(e, str) else e) for e in events]
+        self._description = description
+        self._status = WebhookStatus(status) if isinstance(status, str) else status
+        self._secret = secret
+        self._creation_timestamp = creation_timestamp
+        self._last_updated_timestamp = last_updated_timestamp
+
+    @property
+    def webhook_id(self) -> str:
+        return self._webhook_id
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def url(self) -> str:
+        return self._url
+
+    @property
+    def events(self) -> list[str]:
+        return self._events
+
+    @property
+    def description(self) -> Optional[str]:
+        return self._description
+
+    @description.setter
```

Comment:
q: where do we use this setter for the description?

Reply:
It's just I forgot to remove it.